dagmc_stats/DagmcQuery.py

```diff
@@ -236,3 +303,268 @@ def calc_area_triangle(self, tris=[]):
+        vertex related data
+
+        outputs
+        -------
+        gc_all : dictionary in the form of vertex : gaussian curvature value
+        of the vertex
+        """
+        verts = self.get_verts()
+        gc_all = {}
+        for vert_i in verts:
+            gc_all[vert_i] = self.__gaussian_curvature(vert_i, tri_vert_data)
+        return gc_all
+
+    def __gaussian_curvature(self, vert_i, tri_vert_data):
+        """Get gaussian curvature value of a vertex
+        Reference: https://www.sciencedirect.com/science/article/pii/
+        S0097849312001203
+        Formula 1
+
+        inputs
+        ------
+            vert_i : vertex entity handle
+            tri_vert_data : numpy structured array that stores the
+            triangle and vertex related data
+
+        outputs
+        -------
+            gc : gaussian curvature value of the vertex
+        """
+        vert_entries = tri_vert_data[tri_vert_data['vert'] == vert_i]
+        sum_alpha_angles = sum(vert_entries['angle'])
+        gc = np.abs(2 * np.pi - sum_alpha_angles)
+        return gc
+
+    def __get_lri(self, vert_i, gc_all, tri_vert_data):
+        """Get local roughness value of a vertex
+        Reference: https://www.sciencedirect.com/science/article/pii/
+        S0097849312001203
+        Formula 2, 3
+
+        inputs
+        ------
+            vert_i : vertex entity handle
+            gc_all : dictionary in the form of vertex : gaussian curvature
+                value of the vertex
+            tri_vert_data : numpy structured array that stores the triangle
+                and vertex related data
+
+        outputs
+        -------
+            Lri : local roughness value of the vertex
+        """
+        DIJgc_sum = 0
+        Dii_sum = 0
+        all_tris = self.dagmc_file._my_moab_core.get_adjacencies(
+                vert_i, 2, op_type=0)
+        meshset_tris = self.get_tris()
+        adj_tris = list(set(all_tris) & set(meshset_tris))
+        vert_j_list = list(self.dagmc_file._my_moab_core.get_adjacencies(
+            adj_tris, 0, op_type=1))
+        vert_j_list.remove(vert_i)
+        for vert_j in vert_j_list:
+            # get tri_ij_list (the list of the two triangles connected to both
+            # vert_i and vert_j)
+            tri_i_list = self.dagmc_file._my_moab_core.get_adjacencies(
+                vert_i, 2, op_type=0)
+            tri_j_list = self.dagmc_file._my_moab_core.get_adjacencies(
+                vert_j, 2, op_type=0)
+            tri_ij_list = list(set(tri_i_list) & set(tri_j_list))
+            # rows with tri value as tri_ij_list[0] or tri_ij_list[1]
+            select_tris = (tri_vert_data['tri'] == tri_ij_list[0]) | \
+                          (tri_vert_data['tri'] == tri_ij_list[1])
+            # rows with vert value not equal to vert_i and not equal to vert_j
+            exclude_verts = (tri_vert_data['vert'] != vert_i) & \
```

Comment:
maybe `vert_filter` instead of `exclude_verts`.  This operation  results in the list of verts that we *will* include.